core/server/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -268,7 +268,14 @@ public void accessBlock(long sessionId, long blockId) throws BlockDoesNotExistEx
   public void commitBlock(long sessionId, long blockId)
       throws BlockAlreadyExistsException, BlockDoesNotExistException, InvalidWorkerStateException,
       IOException, WorkerOutOfSpaceException {
-    mBlockStore.commitBlock(sessionId, blockId);
+    // NOTE: this may be invoked multiple times due to retry on client side.
+    // TODO(binfan): find a better way to handle retry logic
+    try {
+      mBlockStore.commitBlock(sessionId, blockId);
```

Comment:
as talked offline, the value of this is to get better chance (maybe small) if the master can be back